django/cantusdb_project/main_app/tests/make_fakes.py

```diff
@@ -293,34 +323,34 @@ def make_fake_source(published=None, title=None, segment=None, segment_name=None
     source = Source.objects.create(
         published=published,
         title=title,
+        segment=segment,
         siglum=siglum,
         rism_siglum=rism_siglum,
+        description=description,
+        summary=summary,
         provenance=provenance,
+        full_source=full_source,
+        indexing_notes=indexing_notes,
         provenance_notes=make_fake_text(
             max_size=MAX_NUMBER_TEXT_CHARS, min_size=MIN_NUMBER_TEXT_CHARS
         ),
-        full_source=full_source,
         date=make_fake_text(SHORT_CHAR_FIELD_MAX),
         cursus=random.choice(cursus_choices),
-        segment=segment,
         source_status=random.choice(source_status_choices),
         complete_inventory=faker.boolean(),
-        summary=summary,
         liturgical_occasions=make_fake_text(
             max_size=MAX_NUMBER_TEXT_CHARS, min_size=MIN_NUMBER_TEXT_CHARS
         ),
-        description=description,
         selected_bibliography=make_fake_text(
             max_size=MAX_NUMBER_TEXT_CHARS, min_size=MIN_NUMBER_TEXT_CHARS
         ),
         image_link=faker.image_url(),
-        indexing_notes=indexing_notes,
         indexing_date=make_fake_text(
             max_size=MAX_NUMBER_TEXT_CHARS, min_size=MIN_NUMBER_TEXT_CHARS
         ),
         json_info=None,
     )
-    source.century.set([make_fake_century()])
+    source.century.set([century])
```

Comment:
Is there a logic behind having some of these things set via `.set()` and some set during the instantiation of the `Source` object? 

It seems that these are the many-to-many fields -- I guess this is to make sure that the associated fake objects are created (ie. that `make_fake_notation()`) is called?

Reply:
Initially, I tried passing `century=century` as an argument to Source.objects.create(), and python gave me error messages, so I followed the pattern I saw in lines 323-326 (pre-modification line numbers). I think you can't just specify a single century because it's a many-to-many field.

Reply:
I added some comments in the changes I pushed a few minutes ago.

> I think all the fake data could be created all in one place and that this would make it easier to follow and/or troubleshoot/modify later.

I can make this change if you'd like. I think it makes sense as it currently is - we handle all the optional parameters by making fake data/objects if certain parameters haven't been supplied, but otherwise, we make the fake objects/data at the moment they're needed, and no earlier.

Maybe this is symptomatic of these `make_fake_<something>` functions being initially defined without any parameters, and slowly accumulating new parameters where they were required by added tests. I think this is true of all the `make_fake`s, and I'm not sure that it's worth refactoring all/any of them.

Reply:
> Maybe this is symptomatic of these `make_fake_<something>` functions being initially defined without any parameters, and slowly accumulating new parameters where required as the test suite evolved. I think this is true of all the `make_fake`s, and I'm not sure that it's worth refactoring all/any of them.

Yes, this has the whiff of that about it. Not ideal, but also, as you say, not worth it to change if this is the way things are.